walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,93 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+
 
 public final class FindMeetingQuery {
+
+  // create a set of events to avoid duplicates
+  private Collection<Event> events = new HashSet<>();
+
+  // the meeting request
+  private MeetingRequest request; 
+
+  private static final Collection<Event> NO_EVENTS = Collections.emptySet();
+  private static final Collection<String> NO_ATTENDEES = Collections.emptySet();
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+ 
+    this.events = events;
+    this.request = request; 
+
+    // Check if duration is longer than a day
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()){
+        return Arrays.asList(); 
+    }
+    // Check for no events or no attendees
+    if (events == NO_EVENTS || request.getAttendees() == NO_ATTENDEES) {
+        return Arrays.asList((TimeRange.WHOLE_DAY)); 
+    }
+    // Arraylist holds event times accordingly     
+    List<TimeRange> mandatory = new ArrayList();
+
+    for (Event event: events) {
+
+        // Check if attendees from event are requested for meeting
+        for (String attendee : event.getAttendees()) {            
+            // Check that attendee is a mandatory attendee
+            if ((request.getAttendees().contains(attendee)) ) {
+                mandatory.add(event.getWhen());
+            }           
+        }
+    }
+
+    // Create empty Arraylist of options
+    List<TimeRange> options = new ArrayList();
+
+    // Check if there are valid events during the day
+    if (mandatory.isEmpty()) {
+        return Arrays.asList((TimeRange.WHOLE_DAY)); 
+    }
+
+    // Sort list of time ranges in ascending order
+    Collections.sort(mandatory, TimeRange.ORDER_BY_START);
+
+    int start = TimeRange.START_OF_DAY;
+    int end = TimeRange.END_OF_DAY; 
+    int lastEventEnd = 0;
+
+    for (int i = 0; i <= (mandatory.size() -1); i++)  {
+        // Get event start and end time
+        int eventStart = mandatory.get(i).start();
```

Comment:
You can use a for-each loop here:
https://docs.oracle.com/javase/8/docs/technotes/guides/language/foreach.html